spicy/toolchain/src/compiler/validator.cc

```diff
@@ -478,10 +479,22 @@ struct VisitorPost : visitor::PreOrder, hilti::validator::VisitorMixIn {
     }
 
     void operator()(spicy::declaration::Hook* n) final {
-        if ( auto field = n->parent<spicy::type::unit::item::Field>() ) {
-            if ( n->isForEach() && ! field->isContainer() )
-                error("'foreach' can only be used with containers", n);
+        int count = 0;
+        switch ( n->hookType() ) {
+            case declaration::hook::Type::ForEach:
+                if ( auto field = n->parent<spicy::type::unit::item::Field>(); field && ! field->isContainer() )
+                    error("'foreach' can only be used with containers", n);
+
+                ++count;
+                break;
+
+            case declaration::hook::Type::Error: ++count; break;
+
+            default: break;
         }
+
+        if ( count > 1 )
+            error("hook cannot have be both types, 'foreach' and '%error'", n);
```

Comment:
This should be impossible to trigger since we do not loop above so we always have `count <= 1`.

Reply:
Yeah, this was a late-stage rework of the logic--which screwed it up.